apollo-federation/src/sources/connect/json_selection/parser.rs

```diff
@@ -712,36 +755,6 @@ impl SubSelection {
     pub fn last_selection_mut(&mut self) -> Option<&mut NamedSelection> {
         self.selections.last_mut()
     }
-
-    // Since we enforce that new selections may only be appended to
-    // self.selections, we can provide an index-based search method that returns
-    // an unforgeable NamedSelectionIndex, which can later be used to access the
-    // selection using either get_at_index or get_at_index_mut.
-    // TODO In the future, this method could make use of an internal lookup
-    // table to avoid linear search.
-    pub fn index_of_named_selection(&self, name: &str) -> Option<NamedSelectionIndex> {
-        self.selections
-            .iter()
-            .position(|selection| selection.name() == name)
-            .map(|pos| NamedSelectionIndex { pos })
-    }
-
-    pub fn get_at_index(&self, index: &NamedSelectionIndex) -> &NamedSelection {
-        self.selections
-            .get(index.pos)
-            .expect("NamedSelectionIndex out of bounds")
-    }
-
-    pub fn get_at_index_mut(&mut self, index: &NamedSelectionIndex) -> &mut NamedSelection {
-        self.selections
-            .get_mut(index.pos)
-            .expect("NamedSelectionIndex out of bounds")
-    }
-}
-
-pub struct NamedSelectionIndex {
-    // Intentionally private so NamedSelectionIndex cannot be forged.
-    pos: usize,
```

Comment:
I removed this code because it's currently unused, and I didn't see a good way to support selections with multiple names under the single-`name` assumption of `SubSelection::index_of_named_selection`.